homeassistant/components/decora_wifi/__init__.py

```diff
@@ -1 +1,100 @@
 """The decora_wifi component."""
+
+import logging
+
+from decora_wifi import DecoraWiFiSession
+from decora_wifi.models.iot_switch import IotSwitch
+from decora_wifi.models.permission import Permission
+from decora_wifi.models.person import Person
+from decora_wifi.models.residence import Residence
+from decora_wifi.models.residential_account import ResidentialAccount
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    session = DecoraWiFiSession()
+    user = await hass.async_add_executor_job(session.login, username, password)
+    if not user:
+        raise ConfigEntryError("could not authenticate")
+
+    hass.data[DOMAIN][entry.entry_id] = session
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    def logout(event: Event) -> None:
+        if session is not None:
+            try:
+                Person.logout(session)
+            except ValueError as err:
+                _LOGGER.error("Failed to log out of myLeviton Service: %s", err)
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, logout)
```

Comment:
Okay so please refresh my mind, I vaguely remember that we HAD to log out.

How should HA handle if you reload the integration? (e.g. `async_unload_entry` is called), if we have to logout, please also pass `logout` to `entry.async_on_unload` to trigger this when unloading the entry (unloading happens when reloading or whenever you remove the integration)